assets/js/common/CheckCustomizationModal/CheckCustomizationInput.jsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from 'react';
+import { noop, capitalize, isBoolean } from 'lodash';
+import Input from '@common/Input';
+import Tooltip from '@common/Tooltip';
+import Label from '@common/Label';
+
+const inputOptions = [
+  { label: 'True', radioValue: true },
+  { label: 'False', radioValue: false },
+];
+
+const renderLabel = (name, defaultCheckValue) => {
+  const formatDefaultValue = (value) =>
+    isBoolean(value) ? capitalize(String(value)) : value;
+
+  const labelContent = (
+    <Label className="block truncate max-w-[200px] sm:max-w-[250px] md:max-w-[300px]">
+      {name}:<br />
+      (Default: {formatDefaultValue(defaultCheckValue)})
+    </Label>
+  );
+
+  return (
+    <Tooltip zIndex="50" content={name} isEnabled={name?.length > 25}>
+      {labelContent}
+    </Tooltip>
+  );
+};
+
+function CheckCustomizationInput({
+  name,
+  defaultCheckValue,
+  currentValue = '',
+  inputIsLocked,
+  inputType,
+  handleInput = noop,
+}) {
+  const [selectedValue, setSelectedValue] = useState(currentValue);
+
+  const handleBooleanChange = (event) => {
+    const newValue = event.target.value === 'true';
+    setSelectedValue(newValue);
+    handleInput(name, newValue);
+  };
+
+  const handleDefaultChange = (inputEvent) => {
+    handleInput(name, inputEvent.target.value);
+  };
+
+  const renderInput = (type) => {
+    switch (type) {
+      case 'boolean':
```

Comment:
and here we can add more cases for what ever scenario we require. What do you think ?

Reply:
Sounds good